src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -889,37 +887,79 @@ void Convolution::createDescriptor(const std::vector<MemoryDescPtr>& inputDesc,
+        DnnlBlockedMemoryDesc memoryDesc(Precision::I32, {stockInputZeroPoints.size()});
+        stockInputZeroPointsMemPtr->Create(memoryDesc, stockInputZeroPoints.data());
+    }
+}
 
-        if (!inputZeroPointsMemPtr) {
-            inputZeroPointsMemPtr.reset(new Memory(getEngine()));
-            DnnlBlockedMemoryDesc memoryDesc(Precision::U8, {inputZeroPoints.size()});
-            inputZeroPointsMemPtr->Create(memoryDesc, inputZeroPoints.data());
+void Convolution::addLegacyZeroPoints(dnnl::primitive_attr& attr) {
+    if (!legacyInputZeroPoints.empty()) {
+        attr.set_input_zero_points(legacyInputZeroPoints.size(), 1 << 1 /*through C dim*/);
+        if (!legacyInputZeroPointsMemPtr) {
+            legacyInputZeroPointsMemPtr.reset(new Memory(getEngine()));
+            DnnlBlockedMemoryDesc memoryDesc(Precision::U8, {legacyInputZeroPoints.size()});
+            legacyInputZeroPointsMemPtr->Create(memoryDesc, legacyInputZeroPoints.data());
         }
     }
 
-    if (!weightsZeroPoints.empty()) {
-        attr.set_weights_zero_points(weightsZeroPoints.size(), 1 << 1 /*through C dim*/);
+    if (!legacyWeightsZeroPoints.empty()) {
+        attr.set_weights_zero_points(legacyWeightsZeroPoints.size(), 1 << 1 /*through C dim*/);
 
-        if (!weightsZeroPointsMemPtr) {
-            weightsZeroPointsMemPtr.reset(new Memory(getEngine()));
-            DnnlBlockedMemoryDesc memoryDesc(Precision::FP32, {weightsZeroPoints.size()});
-            weightsZeroPointsMemPtr->Create(memoryDesc, weightsZeroPoints.data());
+        if (!legacyWeightsZeroPointsMemPtr) {
+            legacyWeightsZeroPointsMemPtr.reset(new Memory(getEngine()));
+            DnnlBlockedMemoryDesc memoryDesc(Precision::FP32, {legacyWeightsZeroPoints.size()});
+            legacyWeightsZeroPointsMemPtr->Create(memoryDesc, legacyWeightsZeroPoints.data());
         }
     }
 
-    if (!outputCompensation.empty()) {
-        attr.set_output_compensations(outputCompensation.size(), 1 << 1 /*through C dim*/);
+    if (!legacyOutputCompensation.empty()) {
+        attr.set_output_compensations(legacyOutputCompensation.size(), 1 << 1 /*through C dim*/);
 
-        if (!outputCompensationMemPtr) {
-            outputCompensationMemPtr.reset(new Memory(getEngine()));
-            DnnlBlockedMemoryDesc memoryDesc(Precision::I32, {outputCompensation.size()});
-            outputCompensationMemPtr->Create(memoryDesc, outputCompensation.data());
+        if (!legacyOutputCompensationMemPtr) {
+            legacyOutputCompensationMemPtr.reset(new Memory(getEngine()));
+            DnnlBlockedMemoryDesc memoryDesc(Precision::I32, {legacyOutputCompensation.size()});
+            legacyOutputCompensationMemPtr->Create(memoryDesc, legacyOutputCompensation.data());
         }
     }
 }
 
+//See the src/plugins/intel_cpu/src/docs/convPostOps.md for details
+void Convolution::SetPostOpsAndZeroPoints(std::vector<dnnl::primitive_attr> &attrs) {
```

Comment:
In general current impl looks reasonable. At the same time it still can be improved. Examples:

- On avx2 systems for Convolution with per-tensor zp or fused quantization we still create 2 versions of attrs. This doesn't make any sense since Brgemm Convs are not supported on AVX2 - only legacy attrs required.
- On avx512 systems for Convolution with per-tensor zp we still create 2 versions of attrs. Obviously this case is supported via Brgemm Convs and legacy attrs are not required.

Both cases should be resolved as part of first inference latency optimization activity. Doesn't block this PR.

Reply:
- On avx2, shouldTryBrgconv should be false except user application enforce this via RTinfo. So most of the avx2 should only use legacy attribute and would not try brgconv attribute.
- On avx512, Yes. We would have per-tensor zp targeting brgconv kernel and  legacy output compensation targeting JIT kernel.  Current brgconv in ONEDNN has unimplement early return with some heuristic code. So have to use legacy attribute as backup if brgconv can't hit. For long-term, maybe we can remove it if onednn remove these limitation. Thx!